gateway-ha/src/main/java/com/lyft/data/gateway/ha/router/RuleReloadingRoutingGroupSelector.java

```diff
@@ -0,0 +1,76 @@
+package com.lyft.data.gateway.ha.router;
+
+import java.io.FileReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashMap;
+import java.util.Optional;
+
+import javax.servlet.http.HttpServletRequest;
+
+import lombok.extern.slf4j.Slf4j;
+import org.jeasy.rules.api.Facts;
+import org.jeasy.rules.api.Rules;
+import org.jeasy.rules.api.RulesEngine;
+import org.jeasy.rules.core.DefaultRulesEngine;
+import org.jeasy.rules.mvel.MVELRuleFactory;
+import org.jeasy.rules.support.reader.YamlRuleDefinitionReader;
+
+@Slf4j
+public class RuleReloadingRoutingGroupSelector
+    implements RoutingGroupSelector  {
+
+  RulesEngine rulesEngine = new DefaultRulesEngine();
+  MVELRuleFactory ruleFactory = new MVELRuleFactory(new YamlRuleDefinitionReader());
+  String rulesConfigPath;
+  Rules rules = new Rules();
+  long lastUpdatedTime;
```

Comment:
These need to be marked [volatile](https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.3.1.4) to allow for consistent updates to be seen across threads.

But also I would recommend using a bit stronger synchronization here? This should not be a hot enough path to be worth optimizing for lock-free read access. You can use a r-w lock or atomics + CAS to get more consistent behavior